packages/fast-components/src/text-field/text-field.template.ts

```diff
@@ -0,0 +1,57 @@
+import { html, when } from "@microsoft/fast-element";
+import { ref } from "@microsoft/fast-element/dist/directives";
+import { TextField } from "./text-field";
+
+export const TextFieldTemplate = html<TextField>`
+    ${when(
+        x => x.childNodes.length,
```

Comment:
I've seen `childNodes.length` being used with `when` in some new PRs. I apparently missed calling this out in an earlier component. This won't work if the child nodes change after the component is initially rendered because child nodes is an exotic, non-array thing coming from native DOM land. Basically, we have no way to automatically observe it for changes. To truly make this work, you would need to base `when` off of a boolean observable and then create a mutation observer to watch the child nodes and update that observable whenever the nodes changed.

I added a new feature to our roadmap to create a `children` decorator that will create a property of observable child nodes and allow you to have a simple callback to respond when they change. That should make this easy in the future.

Reply:
I think label tends to not be dynamic. With that said, we should update this based on the roadmap - I'd prefer to handle these all at once though as it will be a common scenario for all these form elements.